open_spiel/python/examples/leduc_nfsp.py

```diff
@@ -94,66 +109,259 @@ def action_probabilities(self, state, player_id=None):
 
     self._obs["current_player"] = cur_player
     self._obs["info_state"][cur_player] = (
-        state.information_state_tensor(cur_player))
+      state.information_state_tensor(cur_player))
     self._obs["legal_actions"][cur_player] = legal_actions
 
     info_state = rl_environment.TimeStep(
-        observations=self._obs, rewards=None, discounts=None, step_type=None)
+      observations=self._obs, rewards=None, discounts=None, step_type=None)
 
     with self._policies[cur_player].temp_mode_as(self._mode):
       p = self._policies[cur_player].step(info_state, is_evaluation=True).probs
     prob_dict = {action: p[action] for action in legal_actions}
     return prob_dict
 
 
+def latest_checkpoint_dir():
+  return os.path.join(FLAGS.checkpoint_dir, "latest")
+
+
+def checkpoint_dir(episode):
+  return os.path.join(FLAGS.checkpoint_dir, str(episode))
+
+
+def list_saved_checkpoints():
+  return list(sorted(int(p) for p in os.listdir(FLAGS.checkpoint_dir) if p != "latest"))
+
+
+def most_recent_checkpoints(n):
+  return list_saved_checkpoints()[-n + 1:-1]
+
+
+def random_checkpoints(n):
+  checkpoints = list_saved_checkpoints()[:-1]
+  if n > len(checkpoints):
+    n = len(checkpoints)
+  return random.sample(checkpoints, n)
+
+
+def stratified_checkpoints(n):
+  checkpoints = list_saved_checkpoints()[:-1]
+  if n > len(checkpoints):
+    n = len(checkpoints)
+  if n == 0:
+    return []
+  elif n == 1:
+    return [checkpoints[0]]
+  else:
+    return [checkpoints[int(i * (len(checkpoints) - 1) / (n - 1))] for i in range(n)]
+
+
+class OpponentDistribution(metaclass=abc.ABCMeta):
+  """
+  Represents a distribution of opponent agents, conditional on a single fixed agent.
+  """
+
+  @abc.abstractmethod
+  def sample(self, fixed_agent: rl_agent.AbstractAgent, fixed_agent_player_id: int) -> List[rl_agent.AbstractAgent]:
+    pass
+
+
+class IndependentOpponentDistribution(OpponentDistribution):
+  """
+  A distribution where each agent is chosen independently from a uniform distribution.
+  """
+
+  def __init__(self, agents: List[List[rl_agent.AbstractAgent]]):
+    """
+    :param agents: List of agents for each player id. Note that player_id is the first dimension.
```

Comment:
Please change the comments to be consistent with the rest of the code. Similarly for the other comments.

See here, for example: https://github.com/deepmind/open_spiel/blob/0c265fbaa56c418788333fdf3bd373ff12620d14/open_spiel/python/algorithms/nfsp.py#L182